src/main/kotlin/com/safehill/kclient/tasks/outbound/UploadOperationImpl.kt

```diff
@@ -41,13 +69,16 @@ public class UploadOperationImpl(
             )
         }
 
-        val globalIdentifier = UUID.randomUUID().toString()
-
         // 2. Create server asset with the details
         //      2.1 If already exists and any recipients call this.share(), otherwise end early
+        val serverAssets = serverProxy.create(listOf(encryptedAsset), outboundQueueItem.groupId, listOf(AssetQuality.LowResolution))
 
         // 3. Upload encrypted Data to S3
 
+        for (index in serverAssets.indices) {
+            serverProxy.upload(serverAssets[index], encryptedAsset, listOf(AssetQuality.LowResolution))
```

Comment:
I think you know already, but the list of versions is determined by the type of operation:
- for uploads with no sharing we use low and hi
- for uploads with sharing we use low and mid, and do hi in the next pipeline cycle

Reply:
Yes, this is just a beginning